readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/monitoring/PrometheusMetrics.java

```diff
@@ -4,51 +4,138 @@
+        registerPartitionCollector(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+        return buildChild(MESSAGES_WRITTEN, eventName, String.valueOf(partition));
+    }
+
+    public static Counter.Child hearbeatsSentCounter(String eventName, int partition) {
+        registerPartitionCollector(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+        return buildChild(HEARTBEATS_SENT, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child currentRunningOffsetsGauge(String eventName, int partition) {
+        registerPartitionCollector(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(CURRENT_RUNNING_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedOffsetGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+        return buildChild(LATEST_COMMITTED_OFFSETS, eventName, String.valueOf(partition));
+    }
+
+    public static Gauge.Child latestCommittedTimestampGauge(String eventName, int partition) {
+        registerPartitionCollector(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        Gauge.Child gauge = buildChild(LATEST_COMMITTED_TIMESTAMPS, eventName, String.valueOf(partition));
+        return new Gauge.Child() {
+            @Override
+            public void set(double val) {
+                gauge.set(Double.max(gauge.get(), val));
+            }
+
+            @Override
+            public double get() {
+                return gauge.get();
+            }
+        };
+    }
+
+    public static void clearCollectors() {
+        REGISTERED_COLLECTORS.forEach((collector, labels) -> {
+            collector.clear();
+            labels.clear();
+        });
+    }
+
+    public static void clearPartitionCollectors(int partition) {
+        REGISTERED_COLLECTORS.forEach((collector, labels) -> labels
+            .stream()
+            .filter(list -> String.valueOf(partition).equals(list.get(PARTITION_LABEL_IDX)))
+            .forEach(list -> collector.remove(list.toArray(new String[0])))
+        );
     }
 
-    public static Counter.Child buildCounterChild(Counter baseCounter, String eventName, int partition) {
-        return baseCounter.labels(eventName, GarmadonReader.getHostname(), PrometheusHttpConsumerMetrics.RELEASE,
-                String.valueOf(partition));
+    public static Map<SimpleCollector<?>, Set<List<String>>> getRegisteredCollectors() {
+        return new HashMap<>(REGISTERED_COLLECTORS);
     }
 
-    public static Counter.Child buildCounterChild(Counter baseCounter, String eventName) {
-        return baseCounter.labels(eventName, GarmadonReader.getHostname(), PrometheusHttpConsumerMetrics.RELEASE);
+    private static <CHILD> void registerPartitionCollector(SimpleCollector<CHILD> collector, String... labels) {
```

Comment:
We already have PartitionedWriter<MESSAGE_KIND> in hdfs reader, very unconventional, dont we ? ;) 
Also I just followed Prometheus client naming, for readability (but checkstyle forced me to set it uppercase) -> public abstract class SimpleCollector<Child>extends Collector {...}
So in the case you point I think I try to follow good readability of code. Unless you think using a single letter would be better, I'll leave it like that.
Otherwise let's make checkstyle rule to enforce single letter.
I'll prefer to leave it like that. What do you think ?

Reply:
Touché :-)

Let's keep it this way :-)